.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,25 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
```

Comment:
Our other CI workflow triggers on:
```
on:
  # Triggers the workflow on push or pull request events but only for the main branch
  push:
    branches: 
      - main
      - release-branch-*
  pull_request:
    branches:
      - '**'
```
These are _nearly_ the same, but I think consistency would be better.